### PR TITLE
added getSignInLink overload

### LINKED ISSUE #1667: [PORT] [OAuth] small oauth fix
> Port this change from botbuilder-dotnet/master branch:
https://github.com/microsoft/botbuilder-dotnet/pull/3333

The existing overload of GetOauthSignInLinkAsync function with a userId as argument isn't working. It was hardcoding directline, passing empty conversations etc. This PR is to fix that.

# Changed projects
* Microsoft.Bot.Builder

### PR DESCRIPTION
**Pull request by mdrichardson**
Fixes #1667 

## Description

**Note: This incorporates [this PR](https://github.com/microsoft/botbuilder-js/pull/1697) (currently unmerged), since so much of this is dependent on it.** I compared against that branch since it would be easier to see changes. Let me know if you want me to re-submit against Master.

## Specific Changes

Add this overload to `botFrameworkAdapter` for [parity with C#](https://github.com/microsoft/botbuilder-dotnet/pull/3333/files#diff-1f0c90c6fa85dfcad121531bcb2092edR815):

```js
public async getSignInLink(context: TurnContext, connectionName: string, oAuthAppCredentials?: AppCredentials, userId?: string, finalRedirect?: string): Promise<string> {
```

## Testing

Added 3 tests. This method was previously untested.

### CODE REVIEW

File: libraries/botbuilder/src/botFrameworkAdapter.ts [hunk 1]
```diff
@@ -584,7 +584,11 @@ export class BotFrameworkAdapter extends BotAdapter implements ICredentialTokenP
      */
     public async getSignInLink(context: TurnContext, connectionName: string): Promise<string>;
     public async getSignInLink(context: TurnContext, connectionName: string, oAuthAppCredentials?: AppCredentials): Promise<string>;
-    public async getSignInLink(context: TurnContext, connectionName: string, oAuthAppCredentials?: AppCredentials): Promise<string> {
+    public async getSignInLink(context: TurnContext, connectionName: string, oAuthAppCredentials?: AppCredentials, userId?: string, finalRedirect?: string): Promise<string> {
```
Comment (Stevenic) [inline libraries/botbuilder/src/botFrameworkAdapter.ts]: The new params should be add to at least one of the other signatures.  In TypeScript when you overload a methods signature the last signature is always hidden intellisense wise.

Comment (mdrichardson) [reply libraries/botbuilder/src/botFrameworkAdapter.ts]: @stevenic Will amend. Forgot about that.

Comment (Stevenic) [reply libraries/botbuilder/src/botFrameworkAdapter.ts]: The other thing you can do is just drop the other two signatures as I don't see that they're adding any value.


---
In reply to: [385957002](https://github.com/microsoft/botbuilder-js/pull/1803#discussion_r385957002) [](ancestors = 385957002)

**Comment by Stevenic** [comment]
>      * @param oAuthAppCredentials AppCredentials for OAuth.

Please add doc comments for the new params

---
Refers to: libraries/botbuilder/src/botFrameworkAdapter.ts:583 in 9fc09bd. [](commit_id = 9fc09bd90793fc3c3361603992d190ce344df754, deletion_comment = False)

File: libraries/botbuilder/src/botFrameworkAdapter.ts (hunk 1 above)
Comment (mdrichardson) [reply libraries/botbuilder/src/botFrameworkAdapter.ts]: @Stevenic Like this?

```js
public async getSignInLink(context: TurnContext, connectionName: string): Promise<string>;
public async getSignInLink(context: TurnContext, connectionName: string, oAuthAppCredentials?: AppCredentials): Promise<string>;
public async getSignInLink(context: TurnContext, connectionName: string, oAuthAppCredentials?: AppCredentials, userId?: string, finalRedirect?: string): Promise<string>;
public async getSignInLink(context: TurnContext, connectionName: string, oAuthAppCredentials?: AppCredentials, userId?: string, finalRedirect?: string): Promise<string> {
```

Comment (mdrichardson) [reply libraries/botbuilder/src/botFrameworkAdapter.ts]: removed

**Pull request merged**